packages/ui/Touchable/tests/Touchable-specs.js

```diff
@@ -175,6 +250,76 @@ describe('Touchable', () => {
 
 			expect(actual).toBe(expected);
 		});
+
+		test('should call onDragStart event', (done) => {
+			const Component = Touchable(DivComponent);
+			const handler = jest.fn();
+			const {rerender} = render(<Component onDrag={() => {}} />);
+
+			const component = screen.getByTestId('component');
+
+			fireEvent.mouseDown(component, {clientX: 10, clientY: 20});
+			act(() => jest.advanceTimersByTime(20));
+
+			rerender(<Component onDrag={() => {}} onDragStart={handler} />);
+
+			jest.runOnlyPendingTimers();
+
+			fireEvent.mouseMove(component, {clientX: 20, clientY: 30});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 30, clientY: 40});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 40, clientY: 50});
+			fireEvent.mouseUp(component, {clientX: 40, clientY: 50});
+
+			expect(handler).toHaveBeenCalled();
+			done();
+		});
+
+		test('should call onDragEnd event', (done) => {
+			const Component = Touchable(DivComponent);
+			const handler = jest.fn();
+			const {rerender} = render(<Component onDrag={() => {}} />);
+
+			const component = screen.getByTestId('component');
+
+			fireEvent.mouseDown(component, {clientX: 10, clientY: 20});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 20, clientY: 30});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 30, clientY: 40});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 40, clientY: 50});
+			rerender(<Component onDrag={() => {}} onDragEnd={handler} />);
+
+			jest.runOnlyPendingTimers();
+
+			fireEvent.mouseUp(component, {clientX: 40, clientY: 50});
+			expect(handler).toHaveBeenCalled();
+			done();
+		});
+
+		test('should call onBlur event', (done) => {
```

Comment:
this test does not hit the blur() functions of drag or hold ==> it does not increase coverage
In order to hit those lines, we need a focusable element.
In touch.js -> **startGesture(),** this.targetHadFocus = this.target === document.activeElement - in our tests this is false
hadleBlur checkd if the component has focus **call('hasFocus'),** before calling the blur function

It is better to delete this test for the moment

Reply:
Right. Thank you!